R/eval-select.R

```diff
@@ -162,7 +162,7 @@ eval_select_impl <- function(x,
   local_data(x)
 
   if (length(include)) {
-    expr <- quo(all_of(include) | !!expr)
+    expr <- quo((all_of(include) & !(!!expr)) | !!expr)
```

Comment:
I think this introduces double evaluation? Does the following work?

```suggestion
    expr <- quo({
      `_tmp` <- !!expr
      (all_of(include) & !all_of(`_tmp`)) | `_tmp`
    })
```

I would not be surprised if it doesn't work. Perhaps we need recursive calls to `eval_select()` here to solve this?

Reply:
Will need a test for double evaluation.

Reply:
Do you think double evaluation is troublesome? I'd expect that most selections would be fast and stateless.

Reply:
I'd expect the same, but double evaluation is still a bug I think.

Reply:
The main way in which trouble could manifest is through predicate functions (side effects like warnings, non trivial times with wide and long data frames).

I can also take a look if you'd like. Worst case we'll just leave a comment about the issue.

Reply:
Yeah, why don't you finish this one off.